src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -207,32 +142,44 @@ var TestRenderer = ReactFiberReconciler({
     hostContext : Object,
     internalInstanceHandle : Object
   ) : TextInstance {
-    var inst = {
-      text : text,
-      rootContainerInstance,
-      toJSON: () => inst.text,
-    };
-    return inst;
+    return {text};
   },
 
   commitTextUpdate(textInstance : TextInstance, oldText : string, newText : string) : void {
     textInstance.text = newText;
   },
 
   appendChild(parentInstance : Instance | Container, child : Instance | TextInstance) : void {
-    parentInstance.appendChild(child);
+    const index = parentInstance.children.indexOf(child);
+
+    if (index !== -1) {
+      parentInstance.children.splice(index, 1);
+    }
+    parentInstance.children.push(child);
   },
 
   insertBefore(
     parentInstance : Instance | Container,
     child : Instance | TextInstance,
     beforeChild : Instance | TextInstance
   ) : void {
-    parentInstance.insertBefore(beforeChild, child);
+    const index = parentInstance.children.indexOf(child);
+    if (index !== -1) {
+      this.children.splice(index, 1);
+    }
+    const beforeIndex = parentInstance.children.indexOf(beforeChild);
+    if (beforeIndex === -1) {
+      throw new Error('This child does not exist.');
+    }
+    parentInstance.children.splice(beforeIndex, 0, child);
   },
 
   removeChild(parentInstance : Instance | Container, child : Instance | TextInstance) : void {
-    parentInstance.removeChild(child);
+    const index = parentInstance.children.indexOf(child);
+    if (index === -1) {
+      throw new Error('This child does not exist.');
```

Comment:
Note sure those checks are important. They would indicate bugs in Fiber.
They exist in `ReactNoop` because it tests Fiber itself but doesn't look necessary here.